### PR TITLE
Update GEP-1364 to Standard implementation

### LINKED ISSUE #1364: GEP: Conditions and Status update
We currently have a lot of outstanding issues related to status, mostly around Conditions, whether it be issues with specific Conditions being hard to understand, or general issues around Conditions. It's just because a lot of the status and Conditions design has grown very organically, what we effectively need is a design pass to make sure everything is consistent. There have been a few looks at this previously, but nothing coordinated across the whole API.

This issue covers a review of all the outstanding status issues, figuring out which ones relate, generating a TODO list, and then writing a GEP to cover all the changes.

Based on my initial quick pass, I think it's likely that this will include changes that will break conformance, but I think the longer we leave this, the worse the changes will be.

I propose making this a blocker for v0.6.0 release, so that we can get the pain out of the way now, and move forward with a consistent view of how our status flows work.

Here's my initial notes from doing a quick issue roundup.

### Rough initial notes to link in relevant issues

In general, we need to make sure we're reusing the same Condition types across different resources, and that they have the same or similar definitions across resources as well.

The biggest review to date was #1111, and there are still some outstanding items from that.

GatewayClass has `Accepted`. This seems well-defined.

Gateway has `Ready`, which has significant issues, see #1156. This probably needs a follow-up to describe the change.

I think we definitely need to have a better description of what `Ready` means, and ensure that the definitions are consistent. @evankanderson has raised that it would be really useful for knative's use case for `Ready` to mean "Ready to pass traffic as soon as it's set", but I don't know the extent to which we can mandate that. We may be stuck with setting a "`Ready` means will be ready in <5sec" or something, but that will require implementation feedback.

We need to move the Listener condition to `Attached` instead of `Detached`, as described in #1110. Having a small set of standard positive-polarity (that is, they are `status: true` if everything is good) Conditions is useful, and similar things should have similar names.

#1112 also needs to be updated post the closing of #1211.

We also need to note if conditions should be added if they are in a happy state. The latest comments on #1315 is talking about this.
Personally, I prefer that conditions are always added, even if they are `Unknown`, but this one needs discussion.

#1362 should probably also be done as part of this.

### PR DESCRIPTION
**Pull request by youngnick**
**What type of PR is this?**
/kind documentation
/kind gep

**What this PR does / why we need it**:
Updates GEP-1364 (#1364) to implemented in the Standard channel.

Fixes #1364.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -75,14 +75,15 @@ nav:
         - geps/gep-1324.md
         - geps/gep-1282.md
       - Implementable:
-        - geps/gep-1364.md
+        - DummyEntry
```
Comment (robscott) [inline mkdocs.yml:78]: I think this was a temporary change for the sake of CI?
```suggestion
        - DummyEntry
```

Comment (youngnick) [reply mkdocs.yml:78]: no, `mkdocs` will permanently crash if the object is empty. Actually, let me see if I can add the empty list instead...

Comment (robscott) [reply mkdocs.yml:78]: Maybe just remove "Implementable" altogether until we have a GEP that moves to that status?

**Comment by robscott** [comment]
Thanks!

/lgtm

**Pull request merged**